homeassistant/components/zwave_js/number.py

```diff
@@ -0,0 +1,75 @@
+"""Support for Z-Wave controls using the number platform."""
+from typing import Callable, List
+
+from zwave_js_server.client import Client as ZwaveClient
+
+from homeassistant.components.number import DOMAIN as NUMBER_DOMAIN, NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DATA_CLIENT, DATA_UNSUBSCRIBE, DOMAIN
+from .discovery import ZwaveDiscoveryInfo
+from .entity import ZWaveBaseEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities: Callable
+) -> None:
+    """Set up Z-Wave Number entity from Config Entry."""
+    client: ZwaveClient = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+
+    @callback
+    def async_add_number(info: ZwaveDiscoveryInfo) -> None:
+        """Add Z-Wave number entity."""
+        entities: List[ZWaveBaseEntity] = []
+        entities.append(ZwaveNumberEntity(config_entry, client, info))
+        async_add_entities(entities)
+
+    hass.data[DOMAIN][config_entry.entry_id][DATA_UNSUBSCRIBE].append(
+        async_dispatcher_connect(
+            hass,
+            f"{DOMAIN}_{config_entry.entry_id}_add_{NUMBER_DOMAIN}",
+            async_add_number,
+        )
+    )
+
+
+class ZwaveNumberEntity(ZWaveBaseEntity, NumberEntity):
```

Comment:
Values have an optional `unit` property in the metadata, perhaps they can be used. Consider for future.